staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests/unions/union_custom_members/docs_test.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package union_custom_members
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func Test(t *testing.T) {
+	st := localSchemeBuilder.Test(t)
+
+	st.Value(&U{M1: &M1{S: "x"}}).ExpectValid()
+	st.Value(&U{M2: &M2{S: "x"}}).ExpectValid()
+
+	st.Value(&U{M1: &M1{S: "x"}, M2: &M2{S: "x"}}).ExpectInvalid(
+		field.Invalid(nil, "{m1, m2}", "must specify exactly one of: `m1`, `m2`"),
```

Comment:
This is an opportunity to revisit our convention on error strings.  For not-very-good reasons we chose things like "field names should be in backquotes" and "literal strings go in single-quotes".

For example an error today:

```
spec.something.field: Forbidden: may only be set when `otherThing` is 'Foo'
```

could become:

```
spec.something.field: Forbidden: may only be set when .otherThing is "Foo"
```

In particular, what's a better way to indicate that "otherThing" is a field at the same level?

```
`otherThing`
.otherThing
spec.something.otherThing
.spec.something.otherThing
```
?